### PR TITLE
ARROW-11911: [Website] Add protobuf vs arrow to FAQ

### PR DESCRIPTION
**Pull request by westonpace**
This question comes up occasionally as people try and learn Arrow and fit it into their mental model so hopefully this will help.

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -174,6 +174,30 @@ and are commonly used together in applications.  Storing your data on disk
...
+in this format.
+
+Protobuf has libraries to do this deserialization but they do not aim for a
+common in-memory format.  A message that has been deserialized by protoc
+generated C# code is not going to have the same representation as one that has
+been deserialized by protoc generated Java code.  You would need to marshal
+the data from one language to the other.
+
+Arrow avoids this but it comes at the cost of increased space.  This means that
+Protobuf is often a better choice for serializing data on the wire.  Just like
```
Comment (wesm) [inline faq.md:196]: Perhaps we should use slightly different language here to soothe the pedants, "Protobuf can be a better choice for serializing certain kinds of data on the wire (like small individual records or sparse data with many optional fields)."

Comment (westonpace) [reply faq.md]: Agreed, I changed to that.

**Comment by westonpace** [comment]
Gentle nudge, might be good to get this merged so we can catch the release.

File: faq.md [hunk 2]
```diff
@@ -174,6 +174,31 @@ and are commonly used together in applications.  Storing your data on disk
 using Parquet and reading it into memory in the Arrow format will allow
 you to make the most of your computing hardware.
 
+#### **What is the difference between Apache Arrow and Google Protobuf?**
```
Comment (nealrichardson) [inline faq.md:177]: I think you should move this to after the discussion of Feather. It interrupts the flow of the discussion from Parquet (a columnar data file format we recommend) to Feather (if there is an Arrow file format, why do you recommend Parquet?), and IIUC "what about protobuf" is more similar of a question to "what about flatbuffers".

**Review by nealrichardson — COMMENTED** [review]
One suggestion; I don't know the details enough to comment on the substance.

**Comment by westonpace** [comment]
Gentle nudge again, I believe this is ready for review/merge.

**Comment by nealrichardson** [comment]
Thanks for the nudges. Since @wesm has already reviewed and there haven't been major changes since then, I'll merge later in the week if there's no objection.

**Review by wesm — APPROVED** [review]
+1. I'll go ahead and merge. Thanks @westonpace!

File: faq.md [hunk 3]
```diff
@@ -211,6 +211,31 @@ the Arrow format to disk prior to the development of the Arrow IPC file format.
...
 retained the "Feather" name and APIs for backwards compatibility.
 
+#### **How does Arrow relate to Protobuf?**
+
+Google's protocol buffers library (Protobuf) is not a "runtime in-memory
+format."  Similar to Parquet, Protobuf's representation is not suitable for
+processing.  Data must be deserialized into an in-memory representation like
+Arrow for processing.
+
+For example, unsigned integers in Protobuf are encoded as varint where each
```
Comment (pitrou) [inline faq.md:221]: "varint" can be cryptic to the reader

File: faq.md [hunk 4]
```diff
@@ -211,6 +211,31 @@ the Arrow format to disk prior to the development of the Arrow IPC file format.
...
+generated C# code is not going to have the same representation as one that has
+been deserialized by protoc generated Java code.  You would need to marshal
+the data from one language to the other.
+
+Arrow avoids this but it comes at the cost of increased space.  Protobuf can
+be a better choice for serializing certain kinds of data on the wire (like
+individual records or sparse data with many optional fields).  Just like
+Parquet this means that Arrow and Protobuf complement each other well.  For
+example, Arrow Flight uses gRPC and Protobuf to transmit data which is then
+deserialized into Arrow format for processing.
```
Comment (pitrou) [inline faq.md:237]: There is a misunderstanding here. Flight does *not* transmit Arrow data using protobuf. Flight uses the IPC stream protocol. It just adds a gRPC-based protobuf-encoded API around it.

Comment (wesm) [reply faq.md:237]: I tried to clarify the misunderstanding in https://github.com/apache/arrow-site/pull/102. Please push additional changes there to improve the exposition if needed

**Pull request merged**